tests/unit/model/models/modelManager.spec.ts

```diff
@@ -69,4 +73,79 @@ describe('ModelManager', () => {
       await expect(createPromise).rejects.toThrow('store-trigger service is not available');
     });
   });
+
+  describe('deleteModel tests', () => {
+    it('should send a delete request to storeTrigger successfully', async () => {
+      const identifier = createUuid();
+      const record = createRecord();
+      const expectedResponse: StoreTriggerResponse = {
+        jobID: randWord(),
+        status: OperationStatus.IN_PROGRESS,
+      };
+
+      catalogMock.getRecord.mockResolvedValue(record);
+      storeTriggerMock.deletePayload.mockResolvedValue(expectedResponse);
+
+      const response = await modelManager.deleteModel(identifier);
+
+      expect(response).toEqual(expectedResponse);
+    });
+
+    it('rejects with AppError if the identifier is not found in catalog', async () => {
+      const identifier = createUuid();
+
+      catalogMock.getRecord.mockResolvedValue(undefined);
+      storeTriggerMock.deletePayload.mockRejectedValue(
+        new AppError('NOT_FOUND', StatusCodes.NOT_FOUND, `Identifier ${identifier} wasn't found on DB`, true)
+      );
+
+      const response = modelManager.deleteModel(identifier);
+
+      await expect(response).rejects.toThrow(`Identifier ${identifier} wasn't found on DB`);
+    });
+
+    it('rejects if productStatus is PUBLISHED', async () => {
+      const identifier = createUuid();
+      const record = createRecord();
+      record.productStatus = RecordStatus.PUBLISHED;
+
+      catalogMock.getRecord.mockResolvedValue(record);
+      storeTriggerMock.deletePayload.mockResolvedValue(
```

Comment:
You don't resolve an error. You should reject an error.

Reply:
ok